magento/magento2
# Log exception when something unexpected goes wrong during the VIES Va…

…t Number validation process.

### Description (*)
Whenever something unexpected goes wrong with the [VIES VAT number validation service](https://ec.europa.eu/taxation_customs/vies/#/vat-validation) (for example: it being down, or having changed url's, or your server being blocked because having send too many requests, ...) no trace of this was being logged. Which made it hard to figure out exactly what the problem is without having to debug things manually.
This PR changes that and makes sure the exception information gets logged in `var/log/exception.log`

### Related Pull Requests


### Fixed Issues (if relevant)
1. Fixes https://github.com/magento/magento2/issues/36065

### Manual testing scenarios (*)
1. In the code [over here](https://github.com/magento/magento2/blob/2.4.5/app/code/Magento/Customer/Model/Vat.php#L47), change the url to `https://example.com/taxation_customs/vies/checkVatService.wsdl` so we can simulate an unexpected error
2. In the Magento backoffice, go to Stores > Configuration > General > General > Store Information
3. For the Country field, pick an EU country (like France)
4. For the VAT Number field, enter something random, like `012345`
5. Press the `Validate VAT Number` button
6. Inspect the `var/log/exception.log` file
7. With this PR applied, it's expected that an entry is created there to give some hints about what goes wrong with the VAT Number validation service 

### Questions or comments
I haven't added unit tests or any other test, there seems to be very little automatic tests that exist for this part in Magento and I don't want to spend the time in to write everything out from scratch, I don't have the energy or time for that, so if this is required, it would be appreciated if an Adobe employee could help out with that.

### Contribution checklist (*)
